### ISSUE TITLE
SIGFPE on one of two computers (macOS)

### ISSUE THREAD

**Issue by bombsimon** [ibmdb/go_ibm_db#75]
- os Version: `macOS 10.15.3`
- cli-driver Version: `v11.1.4.5 (special_39510/64-bit)`
- Db2 Server Version: n/a
- go Version: `1.13.4` and `1.14`

<details><summary><code>go env</code> Output:</summary><br><pre>
This is the same for both machines so I chose one.
$ go env
AR="ar"
CC="clang"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_ENABLED="1"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
CXX="clang++"
GCCGO="gccgo"
GO111MODULE="auto"
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/simon/Library/Caches/go-build"
GOENV="/Users/simon/Library/Application Support/go/env"
GOEXE=""
GOFLAGS=""
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/q3/bf76j42160j2nt4s2tz1_fd40000gn/T/go-build749845321=/tmp/go-build -gno-record-gcc-switches -fno-common"
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOINSECURE=""
GOMOD="/Users/simon/tmp/db2-test/go.mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="darwin"
GOPATH="/Users/simon/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
PKG_CONFIG="pkg-config"

</pre></details>

Steps to Reproduce:

```sh
#!/bin/bash

set -eux

test_path=db2-test

# Unset environment to re-install, remove existing test (if any)
unset CGO_CFLAGS CGO_LDFLAGS LD_LIBRARY_PATH DYLD_LIBRARY_PATH DB2HOME
rm -rf "$test_path"

# Create and enter test directory
mkdir -p "$test_path"
cd "$test_path"

# Clone the source code for the driver
git clone https://github.com/ibmdb/go_ibm_db.git

# Download and install the cli driver
go run go_ibm_db/installer/setup.go

# Setup environment
export DB2HOME="$(pwd)/clidriver"
export DYLD_LIBRARY_PATH="$(pwd)/clidriver/lib"
export CGO_LDFLAGS="-L$(pwd)/clidriver/lib"
export CGO_CFLAGS="-I$(pwd)/clidriver/include"

# Create go project
go mod init db2test
go get -u github.com/ibmdb/go_ibm_db

# Create a test program to compile
cat > main.go << EOF
package main

import (
    "database/sql"
    "fmt"

    _ "github.com/ibmdb/go_ibm_db"
)

func main() {
    con := "HOSTNAME=localhost;PORT=50000;DATABASE=some_db;UID=xxx;PWD=xxx"

    db, err := sql.Open("go_ibm_db", con)
    if err != nil {
        fmt.Println(err)
    }

    defer db.Close()

    fmt.Println("at least I ran!")
}
EOF

# Build and run the script
go build main.go -o main
./main
```

On one of the machines I can run and compile this with both `1.13.4` and `1.14`.

```text
+ go build main.go
+ ./main
at least I ran!
```

One the other machine (running `1.13`), I get `SIGFPE`.

```text
fatal error: unexpected signal during runtime execution
[signal SIGFPE: floating-point exception code=0x7 addr=0x442180f pc=0x442180f]
runtime stack:
runtime.throw(0x4109a0b, 0x2a)
    /usr/local/Cellar/go/1.13.4/libexec/src/runtime/panic.go:774 +0x72
runtime.sigpanic()
    /usr/local/Cellar/go/1.13.4/libexec/src/runtime/signal_unix.go:378 +0x47c
goroutine 1 [syscall, locked to thread]:
runtime.cgocall(0x40c0c80, 0xc000057e38, 0x1)
    /usr/local/Cellar/go/1.13.4/libexec/src/runtime/cgocall.go:128 +0x5b fp=0xc000057e08 sp=0xc000057dd0 pc=0x40047ab
github.com/ibmdb/go_ibm_db/api._Cfunc_SQLAllocHandle(0x40d0001, 0xc00001808c, 0x4000000)
    _cgo_gotypes.go:202 +0x4c fp=0xc000057e38 sp=0xc000057e08 pc=0x40afc8c
github.com/ibmdb/go_ibm_db/api.SQLAllocHandle(...)
    /Users/simon/go/pkg/mod/github.com/ibmdb/go_ibm_db@v0.2.0/api/zapi_unix.go:21
github.com/ibmdb/go_ibm_db.initDriver(0x0, 0x0)
    /Users/simon/go/pkg/mod/github.com/ibmdb/go_ibm_db@v0.2.0/driver.go:27 +0x6b fp=0xc000057ed0 sp=0xc000057e38 pc=0x40b5cdb
github.com/ibmdb/go_ibm_db.init.0()
    /Users/simon/go/pkg/mod/github.com/ibmdb/go_ibm_db@v0.2.0/driver.go:53 +0x22 fp=0xc000057f00 sp=0xc000057ed0 pc=0x40b5f82
runtime.doInit(0x41bb200)
    /usr/local/Cellar/go/1.13.4/libexec/src/runtime/proc.go:5222 +0x8a fp=0xc000057f30 sp=0xc000057f00 pc=0x403a39a
runtime.doInit(0x41b9a40)
    /usr/local/Cellar/go/1.13.4/libexec/src/runtime/proc.go:5217 +0x57 fp=0xc000057f60 sp=0xc000057f30 pc=0x403a367
runtime.main()
    /usr/local/Cellar/go/1.13.4/libexec/src/runtime/proc.go:190 +0x1da fp=0xc000057fe0 sp=0xc000057f60 pc=0x402e3ca
runtime.goexit()
    /usr/local/Cellar/go/1.13.4/libexec/src/runtime/asm_amd64.s:1357 +0x1 fp=0xc000057fe8 sp=0xc000057fe0 pc=0x4058411
```

As far as I know this occurs due to division/modulo by zero or some integer overflow. Thus, this should behave the same on both machines but it doesn't. Any help is much appreciated!

**Comment by akhilravuri1** [comment]
Hi @bombsimon 

Are you trying to run the same build used for 1.14?

Thanks,
Akhil

**Comment by bombsimon** [comment]
Hi! Thanks for a quick response!

No, as you see in my steps to re-produce I re-build the program each time. So I build it and run it with the same version of go, no moving binaries around.

**Comment by akhilravuri1** [comment]
Thank you.
If you run the file using "go run main.go" are you facing this issue?

**Comment by bombsimon** [comment]
Funny you asked, I tried to run with `go run` previously but never got that to work, see https://github.com/ibmdb/go_ibm_db/issues/24#issuecomment-597708409.

** Working machine**

```sh
$ go run main.go
dyld: Library not loaded: libdb2.dylib
  Referenced from: /var/folders/q3/bf76j42160j2nt4s2tz1_fd40000gn/T/go-build915540258/b001/exe/main
  Reason: image not found
signal: abort trap
```

**Non working machine**

```sh
fatal error: unexpected signal during runtime execution
[signal SIGFPE: floating-point exception code=0x7 addr=0x43d780f pc=0x43d780f]
```

This is actually new to me, never tried `go run` on the second (non working) machine after I discovered I could build and run the binary.

**Comment by akhilravuri1** [comment]
Can you please try installing and setting up the driver using Readme?

without batch file or go mod.

> `go env` Output:

Because I see CGO_CFLAGS are empty.

Thanks,
Akhil

**Comment by bombsimon** [comment]
Oh, my bad! I copied the `go env` from the wrong shell. If you look at my setup process I **definitely** set the C flags. This is the proper output (based on the example above):

```sh
$ go env | grep CGO
GCCGO="gccgo"
CGO_ENABLED="1"
CGO_CFLAGS="-I/Users/simon/tmp/db2-test/clidriver/include"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-L/Users/simon/tmp/db2-test/clidriver/lib"
```

I will have to use this with go modules for my final project, but of course I can test now to troubleshoot! But remember, I get it to compile and run on one machine with the way I set it up. And the machine where it's not working doesn't seem to be related to the setup process, if so I would expect not finding headers ot compilation issues.

This is how I did a clean setup this time, without modules, and exactly according to the README.

```sh
#!/bin/bash

set -eux

test_path=db2-test-no-gomod

unset CGO_CFLAGS CGO_LDFLAGS LD_LIBRARY_PATH DYLD_LIBRARY_PATH DB2HOME
rm -rf "$test_path"

# Create and enter test directory
mkdir -p "$test_path"
cd "$test_path"

# Ensure we never use modules
export GO111MODULE=off

# --- The following steps are faken from the README of the project

# Download driver to ~/go/src
go get -d github.com/ibmdb/go_ibm_db

# Install driver
cd "$GOPATH/src/github.com/ibmdb/go_ibm_db/installer"
go run setup.go

# Setup environment
export DB2HOME=/Users/simon/go/src/github.com/ibmdb/go_ibm_db/installer/clidriver
export CGO_CFLAGS=-I$DB2HOME/include
export CGO_LDFLAGS=-L$DB2HOME/lib
export DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$DB2HOME/lib

# --- The README setup ends here.

# Go back to test folder
cd -

# Create test program
cat > main.go << EOF
package main

import (
	"database/sql"
	"fmt"

	_ "github.com/ibmdb/go_ibm_db"
)

func main() {
	con := "HOSTNAME=localhost;PORT=50000;DATABASE=some_db;UID=xxx;PWD=xxx"

	db, err := sql.Open("go_ibm_db", con)
	if err != nil {
		fmt.Println(err)
	}

	defer db.Close()

	fmt.Println("at least I ran!")
}
EOF
```

After this, my environment looks like this:

```sh
$ for env in $DB2HOME $CGO_CFLAGS $CGO_LDFLAGS $DYLD_LIBRARY_PATH; do echo $env; done
/Users/simon/go/src/github.com/ibmdb/go_ibm_db/installer/clidriver
-I/Users/simon/go/src/github.com/ibmdb/go_ibm_db/installer/clidriver/include
-L/Users/simon/go/src/github.com/ibmdb/go_ibm_db/installer/clidriver/lib
:/Users/simon/go/src/github.com/ibmdb/go_ibm_db/installer/clidriver/lib

$ go env | egrep 'CGO_(LD|C)FLAGS'
CGO_CFLAGS="-I/Users/simon/go/src/github.com/ibmdb/go_ibm_db/installer/clidriver/include"
CGO_LDFLAGS="-L/Users/simon/go/src/github.com/ibmdb/go_ibm_db/installer/clidriver/lib"
```

And trying `go run`:

```sh
$ go run main.go
dyld: Library not loaded: libdb2.dylib
  Referenced from: /var/folders/q3/bf76j42160j2nt4s2tz1_fd40000gn/T/go-build081777954/b001/exe/main
  Reason: image not found
signal: abort trap
```

And the way I got it to work with the original installation:

```sh
$ go build main.go && ./main; echo $?                                                                                                                    
at least I ran!
0```

So I think it's safe to say that it's not my setup process causing me to not be able to execute `go run`.

I will try the setup without go modules and installing the driver in `$GOPATH` although I can't see why this would differ since

* I got the driver and headers
* My environment is correct according to README
* go modules doesn't affect the Go code
* I can compile the program on both machines (and run it on one)

I'll get back with the results from the non working machine but don't expect any different result.

**Comment by bombsimon** [comment]
Wow... I thought I knew what I was doing but either I'm stupid or something is funky with this driver and the way you have to install it. Something with relative pahts?

With the other setup I get yet another result. So, on the machine where I didn't get it to work with the clidriver installed outside of this project I can now execute `go run` and `go build main.go && ./main` with another error. Interestingly this seems to be related to the `SIGFPE` since it's referencing floating point numbers.

```text
# github.com/ibmdb/go_ibm_db/api
../../go/src/github.com/ibmdb/go_ibm_db/api/api_unix.go:102:25: could not determine kind of name for C.SQL_C_DECFLOAT
```

**Comment by akhilravuri1** [comment]
This is related to my latest commit DEC_FLOAT...I will change it.

Give me some time.

Thanks,
Akhil

**Comment by bombsimon** [comment]
Thanks, much appreciated! I am impressed with your quick response to this!

Do you have any idea why I cannot perform `go run` on one of the machines but the same machine can build and run the compiled program? Since the setup process is exactly the same I don't understand why the one showing no issues so far cannot find `libdb2.dylib`.

Also, can you deny or confirm that the `clidriver` requires a relative path to the go library? If so, this might be why it's "working" (not causing the `SIGFPE` panic) when I'm installing the driver in the same folder as where the go code is located. If this is the case I definitely think this should be seen as something that should be fixed, I should be able to move the `clidriver` folder to any desired path as long as my environment points to said path.

I guess most projects do use modules now since it's been around since go 1.11 and thus it's not really suitable to install the driver in the `pkg/mod` folder where the go library will be installed.

**Comment by akhilravuri1** [comment]
I am also trying to replicate this issue.

we run Jenkins's job every day on all platforms with different versions. All got passed.

The only difference is we don't use go mod. we use only go get (JUST README SETPS).

Thanks,
Akhil

**Comment by bombsimon** [comment]
I'm not sure if this is relevant or not but I tried to checkout 5266ca4 which is before the implementation of `DECFLOAT` and now I get this error on the machine not working:

```sh
$ go run main.go
dyld: Library not loaded: libdb2.dylib
  Referenced from: /var/folders/cg/g9sxx0z12x37g3dxq2sl6ylr0000gn/T/go-build620423114/b001/exe/main
  Reason: image not found
signal: abort trap
```

Which is the same as the working one, however it still doesn't work when trying to compile and run it which is yet another error path not seen before.

```sh
$ go build main.go && ./main
dyld: Library not loaded: libdb2.dylib
  Referenced from: /Users/ashkanfathi/tmp/db2-test-no-gomod/./main
  Reason: image not found
Abort trap: 6
```

**Comment by akhilravuri1** [comment]
I have fixed SQL_C_DECFLOAT issue and I have pushed it.

> sure if this is relevant or not but I tried to checkout [5266ca4](https://github.com/ibmdb/go_ibm_db/commit/5266ca44206fe30d3eea7f914d1a5bd862be2762) which is before the

Since you tried with the above commit I guess you will face the same issue. I will try to reproduce this issue and I will come back to you.

Thanks,
Akhil

**Comment by bombsimon** [comment]
Thanks, much appreciated!

Since I personally can only reproduce (or well... produce) this on one of my machines I'm not sure exactly how to do this. The only thing that seems to be consistent between the machines now is that `go run` results in not finding `libdb2.dylib` but other than that one is working and one is not.

If I find something that makes this even easier to reproduce I'll get back to you as well. For now all I know is that given the old commit i can compile and run the binary on one machine and the other one fails with the missing `libdb2.dylib`. Both machines fails with this error while trying `go run`.

**Comment by bombsimon** [comment]
Sorry to spam you, I found out I had a bad `DYLD_LIBRARY_PATH` on the machine not working. So now I unset my whole environment, removed the library and  reinstalled the latest version from `master` with `c_decfloat` removed. I also reinstalled the clidrivers. Everything exactly like the README with the `go get -d`, installing the driver in the `setup` folder, exporting variables to `$GOPATH/go/src/github.com/ibmdb`, not using go modules etcetera.

The working machine is in the same state, that is I can execute `go build main.go && ./main` but I cannot execute `go run main.go`.

The non working machine is also in the same original state where it panics both on `go run` and `go build main.go && ./main`.

```sh
fatal error: unexpected signal during runtime execution
[signal SIGFPE: floating-point exception code=0x7 addr=0x441e80f pc=0x441e80f]
runtime stack:
runtime.throw(0x4108530, 0x2a)
    /usr/local/Cellar/go/1.13.4/libexec/src/runtime/panic.go:774 +0x72
runtime.sigpanic()
    /usr/local/Cellar/go/1.13.4/libexec/src/runtime/signal_unix.go:378 +0x47c
goroutine 1 [syscall, locked to thread]:
runtime.cgocall(0x40bf970, 0xc000052e00, 0x1)
    /usr/local/Cellar/go/1.13.4/libexec/src/runtime/cgocall.go:128 +0x5b fp=0xc000052dd0 sp=0xc000052d98 pc=0x40050fb
github.com/ibmdb/go_ibm_db/api._Cfunc_SQLAllocHandle(0x40d0001, 0xc00009e014, 0x6200000)
    _cgo_gotypes.go:203 +0x4c fp=0xc000052e00 sp=0xc000052dd0 pc=0x40ae85c
github.com/ibmdb/go_ibm_db/api.SQLAllocHandle(...)
    /Users/simon/go/src/github.com/ibmdb/go_ibm_db/api/zapi_unix.go:23
github.com/ibmdb/go_ibm_db.initDriver(0x0, 0x0)
    /Users/simon/go/src/github.com/ibmdb/go_ibm_db/driver.go:28 +0x6b fp=0xc000052e98 sp=0xc000052e00 pc=0x40b490b
github.com/ibmdb/go_ibm_db.init.0()
    /Users/simon/go/src/github.com/ibmdb/go_ibm_db/driver.go:62 +0x4c fp=0xc000052f00 sp=0xc000052e98 pc=0x40b4bdc
runtime.doInit(0x41b8200)
    /usr/local/Cellar/go/1.13.4/libexec/src/runtime/proc.go:5222 +0x8a fp=0xc000052f30 sp=0xc000052f00 pc=0x403a8aa
runtime.doInit(0x41b6a40)
    /usr/local/Cellar/go/1.13.4/libexec/src/runtime/proc.go:5217 +0x57 fp=0xc000052f60 sp=0xc000052f30 pc=0x403a877
runtime.main()
    /usr/local/Cellar/go/1.13.4/libexec/src/runtime/proc.go:190 +0x1da fp=0xc000052fe0 sp=0xc000052f60 pc=0x402e8da
runtime.goexit()
    /usr/local/Cellar/go/1.13.4/libexec/src/runtime/asm_amd64.s:1357 +0x1 fp=0xc000052fe8 sp=0xc000052fe0 pc=0x4058921
```

As you can see it's referencing the non go mod path in `src`. And this is my environment:

```sh
$ for env in $DB2HOME $CGO_CFLAGS $CGO_LDFLAGS $DYLD_LIBRARY_PATH; do echo $env; done
/Users/simon/go/src/github.com/ibmdb/go_ibm_db/installer/clidriver
-I/Users/simon/go/src/github.com/ibmdb/go_ibm_db/installer/clidriver/include
-L/Users/simon/go/src/github.com/ibmdb/go_ibm_db/installer/clidriver/lib
/Users/simon/go/src/github.com/ibmdb/go_ibm_db/installer/clidriver/lib
```

```sh
$ go env | egrep 'CGO_(LD|C)FLAGS'
CGO_CFLAGS="-I/Users/simon/go/src/github.com/ibmdb/go_ibm_db/installer/clidriver/include"
CGO_LDFLAGS="-L/Users/simon/go/src/github.com/ibmdb/go_ibm_db/installer/clidriver/lib"
```

So sadly, back to square one with the panic. Also, not sure if the revert of `c_decfloat` was necessary. I don't know how I got the `C. SQL_C_DECFLOAT` error message and now I cannot reproduce it.

**TL;DR**:
* Old and new release of this library result in the same way as the original post
  * Can compile and execute but not `go run` on the working machine
  * Cannot compile and execute nor `gor run` on the bad machine (panics)
* Installing in $GOPATH/src according to README or with go mod has no effect
* Installing the clidriver in `go_ibm_db/setup` or any other path has no effect

**Comment by tobiassjosten** [comment]
Colleague of @bombsimon here! Thanks for all the help so far, @akhilravuri1.

I was able to reproduce this error on my computer too, with go1.13.7 darwin/amd64. Upgraded to 1.14 but the error persists:

```
fatal error: unexpected signal during runtime execution
[signal SIGFPE: floating-point exception code=0x7 addr=0x43e480f pc=0x43e480f]

runtime stack:
runtime.throw(0x410162b, 0x2a)
        /usr/local/Cellar/go/1.14/libexec/src/runtime/panic.go:1112 +0x72
runtime.sigpanic()
        /usr/local/Cellar/go/1.14/libexec/src/runtime/signal_unix.go:671 +0x46a

goroutine 1 [syscall, locked to thread]:
runtime.cgocall(0x40c0780, 0xc000047e80, 0x41c9801)
        /usr/local/Cellar/go/1.14/libexec/src/runtime/cgocall.go:133 +0x5b fp=0xc000047e50 sp=0xc000047e18 pc=0x400446b
github.com/ibmdb/go_ibm_db/api._Cfunc_SQLAllocHandle(0x40d0001, 0xc00001c08c, 0xc000010000)
        _cgo_gotypes.go:203 +0x4c fp=0xc000047e80 sp=0xc000047e50 pc=0x40afd2c
github.com/ibmdb/go_ibm_db/api.SQLAllocHandle(...)
        /Users/tobiassjosten/.go/src/github.com/ibmdb/go_ibm_db/api/zapi_unix.go:23
github.com/ibmdb/go_ibm_db.initDriver(0x0, 0x0)
        /Users/tobiassjosten/.go/src/github.com/ibmdb/go_ibm_db/driver.go:28 +0x69 fp=0xc000047ee8 sp=0xc000047e80 pc=0x40b5cd9
github.com/ibmdb/go_ibm_db.init.0()
        /Users/tobiassjosten/.go/src/github.com/ibmdb/go_ibm_db/driver.go:62 +0x48 fp=0xc000047f28 sp=0xc000047ee8 pc=0x40b5f48
runtime.doInit(0x41b6480)
        /usr/local/Cellar/go/1.14/libexec/src/runtime/proc.go:5414 +0x8a fp=0xc000047f58 sp=0xc000047f28 pc=0x403ff6a
runtime.doInit(0x41b4bc0)
        /usr/local/Cellar/go/1.14/libexec/src/runtime/proc.go:5409 +0x57 fp=0xc000047f88 sp=0xc000047f58 pc=0x403ff37
runtime.main()
        /usr/local/Cellar/go/1.14/libexec/src/runtime/proc.go:190 +0x1ce fp=0xc000047fe0 sp=0xc000047f88 pc=0x403381e
runtime.goexit()
        /usr/local/Cellar/go/1.14/libexec/src/runtime/asm_amd64.s:1373 +0x1 fp=0xc000047fe8 sp=0xc000047fe0 pc=0x405fa61
exit status 2
```

My `go env`:

```
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/tobiassjosten/Library/Caches/go-build"
GOENV="/Users/tobiassjosten/Library/Application Support/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOINSECURE=""
GONOPROXY=""
GONOSUMDB=""
GOOS="darwin"
GOPATH="/Users/tobiassjosten/.go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/Cellar/go/1.14/libexec"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/Cellar/go/1.14/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-I/Users/tobiassjosten/.go/src/github.com/ibmdb/go_ibm_db/installer/clidriver/include"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-L/Users/tobiassjosten/.go/src/github.com/ibmdb/go_ibm_db/installer/clidriver/lib"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/7q/wrk9fthd2kqgz1p_5hsk5ynw0000gn/T/go-build719327875=/tmp/go-build -gno-record-gcc-switches -fno-common"
```

Any idea how we could proceed to debug this and figure out what's happening?

**Comment by bombsimon** [comment]
Thanks @tobiassjosten!

Since I'm now one of three at the office than can make this compile I had to test this on another machine so I started a VM in GCP I had running Ubuntu 18.04 and had go 1.13.5 installed. The only thing I added in my script above to set this up was `export LD_LIBRARY_PATH=$DB2HOME/lib`.

This is working for me as well, this time even with `go run`.

```sh
$ go run main.go
at least I ran!
```

**Comment by bombsimon** [comment]
@akhilravuri1 I think we finally found the root cause.

It was only one of my machines running 10.15 (Catalina), the other one was still on High Sierra (10.13.6). Apparently, @tobiassjosten was also on 10.13.

I did an OS upgrade on the one not working and lo and behold, it compiles and runs! Not sure if it's worth troubleshooting to get older OSes working, but if not it might be worth mentioning somewhere in the README. Sadly I didn't have the opportunity to test this on 10.14 or versions prior to 10.13.

I won't close this issue with this comment since the problem is still present in 10.13 but if you don't have any plans to support that OS, please feel free to close this issue.

Thank you so much for your time and sorry for the inconvenience where you potentially had to revert the change that wasn't really related to this.

**Comment by akhilravuri1** [comment]
Hi @bombsimon 

Can please try it using the latest cli driver which can be download by using go run setup.go command as stated in the readme file?

Thanks,
Akhil

**Comment by bombsimon** [comment]
@akhilravuri1 Thanks for keeping this open!

Sadly I don't have access to a computer with macOS 10.13 anymore. @tobiassjosten are you still on 10.13? Can you give this a try?

If not I think we should close this issue.

**Comment by tobiassjosten** [comment]
Sorry, I'm afraid I have no way to reproduce this.

**Comment by bombsimon** [comment]
@akhilravuri1 Feel free to close this issue. macOS 10.13 High Sierra will be discontinued 30th November 2020 so it's just a few months away. Thank you for your help looking into this!
